holoviews/plotting/mpl/raster.py

```diff
@@ -82,6 +85,12 @@ def get_data(self, element, ranges, style):
         style['extent'] = [l, r, b, t]
         style['origin'] = 'upper'
 
+        plot_opts = element.opts.get('plot', 'matplotlib')
+        nodata = plot_opts.kwargs.get('nodata')
+        if nodata is not None:
+            data = data if (data.dtype.kind  == 'f') else data.astype(np.float64)
+            data[data == nodata] = np.NaN
```

Comment:
I thought you weren't going to apply `nodata` to float arrays? Or are you just not going to suggest applying it?

Reply:
We decided to simply ignore floats. Implemented in f146bd3